src/string/builder.cr

```diff
@@ -61,6 +61,29 @@ class String::Builder
     @bytesize == 0
   end
 
+  # Chomps the last byte from the string buffer.
+  # If the byte is '\n' and there's a '\r' before it, it is
+  # also removed.
+  def chomp!(byte : UInt8)
+    if bytesize > 0 && buffer[bytesize - 1] == byte
+      back(1)
+
+      if byte === '\n' && bytesize > 0 && buffer[bytesize - 1] === '\r'
```

Comment:
@asterite I think that's confusing behaviour personally, and I can imagine cases where you would want to chomp `\n` and not `\r\n`. If you want to chomp both `\n` and `\r\n` why not just use chomp with no args? Copying ruby here seems ugly.